openquake/hazardlib/site.py

```diff
@@ -577,6 +577,24 @@ def assoc(self, site_model, assoc_dist, ignore=()):
                                  'check the site model' % param)
         return site_model
 
+    def extend(self, lons, lats):
+        """
+        Extend the site collection to additional (and different) points.
+        Used for station_data in conditioned GMFs.
+        """
```

Comment:
TODO: add an assert, the extra points must not have intersections with the original points (at 5 digits)